website/docs/index.md

```diff
@@ -53,6 +53,41 @@ ln -s -f ../../scripts/pre-commit .git/hooks/pre-commit
 ```
 
 
+## How to Create a Good UNIX Interface
+
+One of our goals with this project is learning how to write programs and, more importantly,
+the importance of not writing code vs. writing code. 
+
+Here are some 
+- [good rules for writing good unix programs](https://lasr.cs.ucla.edu/ficus-members/geoff/interfaces.html)
+
+A good example can be seen in notions of progress bars. The idea of progress bars has come up twice now, from 
+two different contributors, once in cp, and once in dd.
```

Comment:
`cp`, a command which copes files has an option for indicating progress, `-v`, after each file is copied.
`cpio`, a command which unpacks files has an option for indicating progress, `-v`, after each file is unpacked.
`flashrom`, a command which writes to flash has an option for indicating progress, `-V`, after each sector is flashed.
Where is the line drawn?
Can `dd`, a command which copies blocks have an option for indicating progress, `-v`, after each block is copied?